File: backend/src/main/kotlin/outcobra/server/service/ExamService.kt
package outcobra.server.service

import outcobra.server.model.dto.ExamDto
import outcobra.server.service.base.BaseService

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
interface ExamService : BaseService<ExamDto> {
    fun readAllBySemester(semesterId: Long): List<ExamDto>
    fun readAll(): List<ExamDto>
    fun readAllInActiveSemesters(): List<ExamDto>
}

File: frontend/src/app/exam/service/exam.service.ts
import {Injectable} from '@angular/core';
import {AppCrudService} from '../../core/services/core/app-crud.service';
import {ExamDto} from '../model/exam.dto';
import {Observable} from 'rxjs';
import {HttpClient} from '@angular/common/http';

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */

@Injectable()
export class ExamService extends AppCrudService<ExamDto> {
    constructor(http: HttpClient) {
        super(http, '/api/exam')
    }

    public readBySemester(semesterId: number): Observable<ExamDto[]> {
        let uri: string = `/semester/${semesterId}${this._baseUri}`;
        return this._http.get<ExamDto[]>(uri);
    }

    public readAllActive(): Observable<ExamDto[]> {
        let uri: string = '/api/active/exam';
        return this._http.get<ExamDto[]>(uri);
    }


}
